opm/parser/eclipse/EclipseState/SimulationConfig/HysteresisOptions.cpp

```diff
@@ -0,0 +1,230 @@
+#include <opm/parser/eclipse/Utility/String.hpp>
+
+namespace Opm {
+
+namespace {
+
+HysteresisOptions::Model readHysteresisModel( int x ) {
+    using Model = HysteresisOptions::Model;
+    switch( x ) {
+        case -1: return Model::Equilibration;
+        case  0: return Model::CarlsonHysteresisNonWettingDrainageWetting;
+        case  1: return Model::CarlsonHysteresisNonWettingImbibitionNonWetting;
+        case  2: return Model::KilloughHysteresisNonWettingDrainageWetting;
+        case  3: return Model::KilloughHysteresisNonWettingImbibitionWetting;
+        case  4: return Model::KilloughHysteresisNonWettingKilloughHysteresisWetting;
+        case  5: return Model::CarlsonNonWettingGasWaterDrainageOil;
+        case  6: return Model::KilloughNonWettingGasWaterDrainageOil;
+        case  7: return Model::KilloughHysteresisNonWettingGasWaterWettingOil;
+        case  8: return Model::JargonHysteresisNonWettingDrainageWetting;
+        case  9: return Model::JargonHysteresisNonWettingImbibitionWetting;
+        default:
+            throw std::invalid_argument( "Hysteresis model must be an integer [-1, 9]" );
+    }         
+}
+
+int readHysteresisApply( const DeckItem& item ) {
+    if( item.defaultApplied( 0 ) ) return 0;
+
+    const auto& key = uppercase( item.getTrimmedString( 0 ) );
+
+    if( key == "BOTH" ) return 0;
+    if( key == "PC" ) return 1;
+    if( key == "KR" ) return 2;
+
+    throw std::invalid_argument( "Limiting hysteresis flag must be 'PC', 'KR' or 'BOTH'" );
+}
+
+HysteresisOptions::CapillaryPressureScanningCurveShape readHysteresisCurveShape( const DeckItem& item ) {
+    if( item.defaultApplied( 0 ) )
+        return HysteresisOptions::CapillaryPressureScanningCurveShape::RETR;
+
+    const auto& key = uppercase( item.getTrimmedString( 0 ) );
+    if( key == "RETR" )
+        return HysteresisOptions::CapillaryPressureScanningCurveShape::RETR;
+
+    if( key == "NEW" )
+        return HysteresisOptions::CapillaryPressureScanningCurveShape::RETR;
+
+    throw std::invalid_argument( "Flag controlling the shape of capillary pressure curves on retraversal must be 'RETR' or 'NEW'" );
+}
+
+bool readHysteresisCorrectImbibition( const DeckItem& item ) {
+    if( item.defaultApplied( 0 ) ) return false;
+
+    const auto& key = uppercase( item.getTrimmedString( 0 ) );
+    if( key == "DRAIN" ) return false;
+    if( key == "BOTH" ) return true;
+
+    throw std::invalid_argument( "Flag controlling fluid mobility correction application on imbibition curves must be 'DRAIN' or 'BOTH'" );
+}
+
+Phase readHysteresisWettingPhase( const DeckItem& item ) {
+    if( item.defaultApplied( 0 ) ) return Phase::OIL;
+
+    const auto& key = uppercase( item.getTrimmedString( 0 ) );
+    if( key == "OIL" ) return Phase::OIL;
+    if( key == "GAS" ) return Phase::GAS;
+
+    throw std::invalid_argument( "Hysteresis Wetting phase must be 'OIL' or 'GAS'" );
+}
+
+bool yesno( const DeckItem& item ) {
```

Comment:
funny function name. maybe `isYesOrNo()`. it is also not specific to the hysteresis stuff, so possibly it makes sense to have this directly in the `DeckItem` class...

Also is it correct to only consider the first character? it currently returns true for strings like `yabba daba doo` or `njet`!